scripts/dudle.coffee

```diff
@@ -54,7 +54,7 @@ module.exports = (robot) ->
     robot.respond /dudlelist/i, (msg) ->
         msg.send 'Aktuell tracke ich folgende Dudles:'
         dudles = read_dudles_file()
-        for dudle in dudles
+        dudles.forEach (dudle) ->
```

Comment:
The problem is that i changed `read_dudles_file` to return a `Map` rather than an `Array`.

As I mentioned in the issue description Coffeescripts `for` loops way work well on arrays and objects but unlike in javascript the `for ... of` loop is not customisable. Meaning that it will always iterate over the properties of an object. In JavaScript `for ... of` can be customised which in case of the map results in it iterating over the keys and values.

I use forEach here because coffee script does not change its semantic.